debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java

```diff
@@ -453,6 +453,7 @@ public void alwaysDispatchHeartbeatEvent(P partition, OffsetContext offset) thro
     public void dispatchHeartbeatEvent(P partition, OffsetContext offset) throws InterruptedException {
         if (heartbeat instanceof ScheduledHeartbeat scheduledHeartbeat) {
```

Comment:
OK, this seems to be a bit of issue wrt the code robustness and maintenance. Is there a way how we could modify the contract/interface in the way we can drop `instanceof`?

Reply:
Yes, we should align all the external connectors (Spanner, Vitess, etc.) to use the factory. I initially thought this could be addressed in a separate PR, but we can apply it in this one.

Reply:
@kmos I am fine doiing it in a separate PR. Let's get the correct behaviour first and refactoring as follow-up.